### PR TITLE
Better types for events

### PR DESCRIPTION
**Pull request by HeadTriXz**
Changed the interfaces to tuples instead, which makes it easier for people to do stuff like `keyof ClientEvents` giving you all event names as a type. 

Also added **once**, **off** and **emit**.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -490,99 +490,96 @@ declare namespace Eris {
...
-    (event: "connect" | "end" | "ready" | "start", listener: () => void): T;
-    (event: "debug" | "warn", listener: (message: string) => void): T;
-    (event: "disconnect", listener: (err?: Error) => void): T;
-    (event: "error", listener: (err: Error) => void): T;
-    (event: "pong", listener: (latency: number) => void): T;
-    (event: "speakingStart" | "speakingStop" | "userDisconnect", listener: (userID: string) => void): T;
-    (event: "unknown", listener: (packet: unknown) => void): T;
+  interface EventListeners {
+    callCreate: [call: Call];
+    callDelte: [call: Call];
```
Comment (aetheryx) [inline index.d.ts]: typo

**Comment by DonovanDMC** [comment]
finally, events as tuples, so I can stop doing all of my custom event stuff manually

**Review by DonovanDMC — APPROVED** [review]

**Comment by coolcalcacol** [comment]
Perhaps also add removeListener for each event?

File: index.d.ts [hunk 2]
```diff
@@ -1837,7 +1833,14 @@ declare namespace Eris {
     unbanGuildMember(guildID: string, userID: string, reason?: string): Promise<void>;
     unpinMessage(channelID: string, messageID: string): Promise<void>;
     validateDiscoverySearchTerm(term: string): Promise<{ valid: boolean }>;
-    on: ClientEvents<this>;
+    emit<K extends keyof ClientEvents>(event: K, ...args: ClientEvents[K]): boolean;
+    emit(event: string, ...args: any[]): boolean;
```
Comment (bsian03) [inline index.d.ts:1837]: Don't think we need `emit` to be typed as the only time it'll be used is within Eris internals

Comment (DonovanDMC) [reply index.d.ts:1837]: I personally use it to reemit interactions for regular command handling (causes less duplicated code, and I don't feel like reinitializing my event listeners)

And what can something like this really hurt? It still allows for the addition of arbitrary events via the last line

Comment (bsian03) [reply index.d.ts:1837]: I feel like for a normal use case, including this seems pretty pointless

Comment (abalabahaha) [reply index.d.ts:1837]: Eh seems ok

**Review by bsian03 — APPROVED** [review]

**Pull request merged**